src/main/java/com/google/sps/servlets/PacketParserServlet.java

```diff
@@ -0,0 +1,167 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License. 
+
+package com.google.sps.servlets;
+
+import com.google.sps.datastore.PCAPdata;
+import com.google.sps.datastore.GenericPCAPDaoImpl;
+
+import io.pkts.PacketHandler;
+import io.pkts.Pcap;
+import io.pkts.buffer.Buffer;
+import io.pkts.packet.Packet;
+import io.pkts.packet.TCPPacket;
+import io.pkts.packet.UDPPacket;
+import io.pkts.packet.IPPacket;
+import io.pkts.protocol.Protocol;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+
+import java.util.*; 
+import com.google.gson.Gson;
+
+import java.io.IOException;
+import java.io.*;
+
+/** Servlet that processes raw PCAP files within the repo. 
+   * We define MYIP to be the IP address of the local machine that was used to create the PCAP file, 
+     and OUTIP to be all of the IP addresses that machine connected with.
+   * allPCAP HashMap indexes all unique OUTIPs in a PCAP file to a PCAPdata object that stores needed attributes, 
+     including the number of connections between MYIP and OUTIP 
+   * We load datastore with all of the unique PCAPdata objects from our HashMap with a given entity tag.
+   * Hard-coded the source IP address (MYIP) -- need to find a way to retrieve MYIP (look at first packet, ask user, use WhatsMyIP?)
+*/
+@WebServlet("/PCAP-data")
+public class PacketParserServlet extends HttpServlet {
+  HashMap<String, PCAPdata> allPCAP = new HashMap<String, PCAPdata>();
+  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService(); //creates database
+
+  //static final String FILENAME = "WEB-INF/files/traffic.pcap";
+  //static final String FILENAME = "WEB-INF/chargen-udp.pcap";
+  //static final String FILENAME = "WEB-INF/files/chargen-tcp.pcap";
+  static final String FILENAME = "WEB-INF/files/smallFlows.pcap";
+
+  static final String MYIP = "192.168.3.131"; // for smallFlows.pcap 
+  static final String ENTITY_TAG = "file_1";
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    parseRaw(FILENAME);
+    putDatastore(ENTITY_TAG);
+  }
+
+  /* Reads PCAP file (from file name) as a stream and puts unique connections into allPCAP HashMap. */
+  public void parseRaw(String file) throws IOException {
```

Comment:
Avoid having these functions in your servlet itself. Create separate class called PCAPParser that takes in the file-name in constructor. That will then have a function parseRaw that returns a list of PCAP objs or creates it. That class can also  have the write PCAP to datastore function.

Reply:
Resolved in latest commits.